tests/tests-common/src/main/java/com/hazelcast/simulator/tests/queue/TxnQueueWithLockTest.java

```diff
@@ -66,14 +66,20 @@ public void timeStep(ThreadState state) {
                     state.counter.failedRollbacks++;
                     logger.severe(name + ": Exception in roll " + state.counter, rollException);
                 }
-            } finally {
-                firstLock.unlock();
             }
         } catch (Exception e) {
             logger.severe(name + ": outer Exception" + state.counter, e);
+        } finally {
+            firstLock.unlock();
         }
     }
 
+    @BeforeRun
```

Comment:
The purpose of this test I'm failing to see. What id the added value of the second lock?